lib/xandra/retry_strategy.ex

```diff
@@ -146,4 +202,63 @@ defmodule Xandra.RetryStrategy do
       end
     end
   end
+
+  @spec run_cluster_with_retrying(Keyword.t(), [{pid(), Host.t()}, ...], (pid() -> result)) ::
+          result
+        when result: var
+  def run_cluster_with_retrying(options, connected_hosts, fun) do
+    [{conn, _host} | _connected_hosts] = connected_hosts
+
+    options =
+      Keyword.merge(options,
+        execution_level: :cluster,
+        connected_hosts: connected_hosts,
+        target_connection: conn
+      )
+
+    case Keyword.pop(options, :retry_strategy) do
+      {nil, _options} ->
+        fun.(conn)
+
+      {retry_strategy, options} ->
+        run_cluster_with_retrying(options, connected_hosts, retry_strategy, fun)
+    end
+  end
+
+  defp run_cluster_with_retrying(options, connected_hosts, retry_strategy, fun) do
+    {conn, options} =
+      case Keyword.pop(options, :target_connection) do
+        {conn, options} when is_pid(conn) ->
+          {conn, options}
+
+        {:random, options} ->
```

Comment:
Why did we add `:random` here? Can't a retry strategy implement it by itself?

Reply:
This is only for internal use, not exposed to the outside. It's basically a fallback for when a `:target_connection` doesn't exist.

Reply:
Not sure whether I understood "implement it by itself"

Reply:
I mean, a retry strategy can implement choosing a random node if `:target_connection` doesn't exist, right? Unless we have a strong reason, let's just remove this.

Reply:
Yes, exactly, we pop the `:target_connection` before calling `retry_strategy.retry`and then do 

```
            |> Keyword.put_new(:target_connection, :random)
```
after calling `retry_strategy.retry`, so that if it is empty, we'll be assigning it a random node. This is to have a mechanism to initially select the first node from load balancing strategy.

Reply:
But using `nil` instead of `:random` also works, as in https://github.com/lexhide/xandra/pull/329/commits/d54ab5bf0bb9ea0b9865e0cdb288f5d5a7422e6e, if that's what you meant